src/services/mentors.js

```diff
@@ -970,42 +963,17 @@ module.exports = class MentorsHelper {
 				}
 			}
 
-			const mentorOrgCode = mentorProfile.data.result.organization_code
-
-			// If no mentor extension found, but user has admin/mentor roles, check if user extension exists and update it
-			if (!mentorExtension && roles && roles.some((role) => role.title === 'admin' || role.title === 'mentor')) {
-				// Try to get user extension without is_mentor filter (unscoped call - cannot cache)
-				const userExtension = await mentorQueries.getMentorExtension(id, [], true, tenantCode)
-				if (userExtension) {
-					// Update using unscoped updateMentorExtension with custom filter to bypass is_mentor constraint
-					await mentorQueries.updateMentorExtension(
-						id,
-						{ is_mentor: true },
-						{},
-						{ user_id: id },
-						true,
-						tenantCode
-					)
-
-					// Cache invalidation: Delete old cache after mentor extension update
-					try {
-						await cacheHelper.mentor.delete(tenantCode, orgCode, id)
-					} catch (cacheError) {
-						console.error(`Cache deletion failed for mentor ${id} after update:`, cacheError)
-					}
-				}
-			}
-
-			if (!mentorProfile.data.result || !mentorExtension) {
+			if (!mentorExtension) {
```

Comment:
Can we move this further up to skip unnecessary processing? Maybe ideally right after the get call.